### ISSUE TITLE
Layers' Validation

### ISSUE THREAD

**Issue by Nada119** [mourra950/Siemens-System-Level-Modelling-of-ASDLA-Graduation-Project#41]
### What feature to add

The Layers' Validation Bug manifests as issues with incorrect parameter initialization, input validation failures, unexpected forward pass outputs, invalid loss values, and NaN values in model parameters during neural network training, impacting model performance and stability.

### Milestone

Milestone 2

### What labels is that issue related to

Pytorch, Enhancement

**Comment by mourra950** [comment]
Successfully Created a system for model analyzing and validation and correction closing the issue
